CountChange2.java

```diff
@@ -0,0 +1,46 @@
+package day8;
+
+import java.util.*;
+
+//Time Complexity : O(mn) where m is the number of coins and n is the amount 
+//Space Complexity : O(n) where n is the length of table
+//Did this code successfully run on Leetcode : no
+//Any problem you faced while coding this : facing problem in conversion of long to int to 
+//											run this solution on leetcode
+
+
+//Your code here along with comments explaining your approach
+public class CountChange2 {
+	static long countWays(int S[], int m, int n) 
+    { 
+        // table[i] will be storing the number of solutions 
+        // for value i. We need n+1 rows as the table is 
+        // constructed in bottom up manner using the base 
+        // case (n = 0) 
+        long[] table = new long[n+1]; 
```

Comment:
Why you made this as long ?
If you want to return int array, then you will need a new array where you can copy all values after type casting.